src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/util/InterruptHandlerState.java

```diff
@@ -203,4 +208,34 @@ public void reset() {
         pendingFinalizationSignals = false;
         semaphoresToSignal.clear();
     }
+
+    public void executeTrigger(final VirtualFrame frame, final SignalSemaphoreNode signalSemaporeNode) {
```

Comment:
This `execute` method is no longer part of a node which is not a good idea. It's probably better to merge `InterruptHandlerState` into `InterruptHandlerNode` (other way around).

Reply:
I could have renamed the method, but the problem remains -see previous comment. Unfortunately we have to deal with global state here